handles/models.py

```diff
@@ -118,14 +119,25 @@ class Meta:
     def __str__(self):
         return f"@{self.name}"
 
-    def clean(self):
+    def _validate_exclusive(self):
+        if (
+            self.has_workspace
+            and self.has_user
+            and self.workspace.created_by_id == self.user.id
+        ):
+            # TODO: remove this once all handles are migrated
+            return
```

Comment:
to handle an in-between state where the same handle can be attached to a personal workspace & its user. won't be needed after `cleanup_handles()` is run in the script